File: 240408/개인정보/personal-info.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.Arrays;

class Person {

    String name;
    int height;
    double weight;

    Person(String name, int height, double weight) {
        this.name = name;
        this.height = height;
        this.weight = weight;
    }
};


public class Main {

    static int N = 5;
    static StringTokenizer st;
    static Person[] people;
    static StringBuilder result = new StringBuilder();

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        people = new Person[N];
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            people[i] = new Person(
                st.nextToken(),
                Integer.parseInt(st.nextToken()), 
                Double.parseDouble(st.nextToken())
            );
        }

        result.append("name").append('\n');
        Arrays.sort(people, (p1, p2) -> {
            return p1.name.compareTo(p2.name);
        });
        for (Person s : people) {
            result.append(String.format("%s %d %.1f\n", s.name, s.height, s.weight));
        }
        result.append('\n');

        result.append("height").append('\n');
        Arrays.sort(people, (p1, p2) -> {
            return p2.height - p1.height;
        });
        for (Person s : people) {
            result.append(String.format("%s %d %.1f\n", s.name, s.height, s.weight));
        }
        System.out.print(result);
    }
}

File: 240418/개인정보/personal-info.py
class Person:
    def __init__(self, n, h, w):
        self.n = n
        self.h = h
        self.w = w

people = []
for _ in range(5):
    n, h, w = input().split()
    h = int(h)
    w = float(w)
    people.append(Person(n, h, w))

print('name')
for s in sorted(people, key=lambda s: s.n):
    print(s.n, s.h, s.w)

print()

print('height')
for s in sorted(people, key=lambda s: -s.h):
    print(s.n, s.h, s.w)